src/ucs/datastruct/mpool.c

```diff
@@ -113,6 +121,15 @@ ucs_status_t ucs_mpool_init(const ucs_mpool_params_t *params, ucs_mpool_t *mp)
         goto err_strdup;
     }
 
+    min_chunk_size = ucs_mpool_chunk_size(mp, 1);
+    if (params->max_chunk_size < min_chunk_size) {
+        ucs_free(mp->data->name);
+        ucs_free(mp->data);
+        mp->data = NULL;
+        ucs_error("Invalid memory pool parameter: max_chunk_size");
+        return UCS_ERR_INVALID_PARAM;
```

Comment:
need to use goto error flow. something like:

```
    ucs_error("invalid memory pool parameter: chunk size is too small (%zu)", 
              params->max_chunk_size);
    status = UCS_ERR_INVALID_PARAM;
    goto err_free_name;
}

...
err_free_name:
     ucs_free(mp->data->name);
err_strdup:
    ucs_free(mp->data);
    mp->data = NULL;
    return status;
```

Reply:
fixed accordingly